docs/orchestrating-elastic-stack-applications/agent-fleet.asciidoc

```diff
@@ -536,23 +538,160 @@ When deployed with ECK, the Fleet Server Pod makes an HTTP call to itself during
 
 Solution: enable CNI when starting minikube: `minikube start --cni=true`.
 
+// tag::elastic-agent-fleet-known-limitations-local-state[]
+
 === Storing local state in host path volume
-Elastic Agent when managed by ECK stores local state in a host path volume. This ensures that integrations run by the agent can continue their work without duplicating work that has already been done after the Pod has been recreated for example because of a Pod configuration change.
-Multiple replicas of an agent, for example Fleet Servers, can not be deployed on the same underlying Kubernetes node as they would try to use the same host path. If local state storage in host path volumes is not desired this can be turned off by configuring an `emptyDir` volume instead:
+Elastic Agent managed by ECK stores local state in a host path volume by default. This ensures that integrations run by the agent can continue their work without duplicating work that has already been done after the Pod has been recreated for example because of a Pod configuration change. Multiple replicas of an agent, for example Fleet Servers, can not be deployed on the same underlying Kubernetes node as they would try to use the same host path. There are 2 options for managing this feature:
+
+1. If local state storage in `hostPath` volumes is not desired this can be turned off by configuring an `emptyDir` volume instead.
 
 [source,yaml]
 ----
+# Using an emptyDir instead of a hostPath volume.
 apiVersion: agent.k8s.elastic.co/v1alpha1
 kind: Agent
 metadata:
   name: fleet-server
 spec:
   deployment:
-    replicas: 3
     podTemplate:
       spec:
+        securityContext: <1>
+          fsGroup: 1000
+          # runAsUser: 0 <2>
         volumes:
         - name: agent-data
           emptyDir: {}
 ...
 ----
+<1> Gid 1000 is the default group at which the Agent container runs. Adjust as necessary if `runAsGroup` has been modified.
+<2> runAsUser: 0 can be disabled when running with an `emptyDir` as opposed to a `hostPath` volume.
+
+[.text-left]
+1. If local state storage is still desired but running the Agent container as root is not allowed, then you can run a `DaemonSet` that adjusts the permissions for the Agent local state on each Node prior to running Elastic Agent. *Note that this `DaemonSet` must be `runAsUser: 0` and possibly `privileged: true`*. *Also note the Kibana changes required to trust the Elasticsearch CA*.
```

Comment:
```suggestion
+
2. If local state storage is still desired but running the Agent container as root is not allowed, then you can run a `DaemonSet` that adjusts the permissions for the Agent local state on each Node prior to running Elastic Agent. *Note that this `DaemonSet` must be `runAsUser: 0` and possibly `privileged: true`*. *Also note the Kibana changes required to trust the Elasticsearch CA*.
```

Reply:
Actually I think these examples below point 2. should be more prominent and I would move them into a new section in the `Configuration` chapter under a title like `Running as a non-root user` and the link to them from here and same for the standalone Agent. This is an important enough new "feature" if you will that we should not hide away in the hostPath limitations section.

Reply:
This new section has been added in both the fleet+agent, and standalone sections.